### PR TITLE
*: use summary log

### PR DESCRIPTION
**Pull request by 3pointer**

### CODE REVIEW

File: pkg/utils/unit.go [hunk 1]
```diff
@@ -8,3 +17,33 @@ const (
 	GB
 	TB
 )
+
+type logCollector struct {
```
Comment (overvenus) [inline pkg/utils/unit.go]: Could you move it to a new package `pkg/summary`?

File: pkg/utils/unit.go [hunk 2]
```diff
@@ -8,3 +17,33 @@ const (
...
+	mu     sync.Mutex
+	fields []zap.Field
+}
+
+// LogCollector collects time into summary log
+var LogCollector *logCollector
```
Comment (overvenus) [inline pkg/utils/unit.go]: We should provide an API to **replace** the default collector which is similar to
 https://godoc.org/google.golang.org/grpc/grpclog#SetLoggerV2

File: pkg/utils/unit.go (hunk 1 above)
Comment (3pointer) [reply pkg/utils/unit.go]: sure, [moved](https://github.com/pingcap/br/pull/118/files#diff-500e3c8287bac6a5d41b94b101065518R13)

File: pkg/utils/unit.go (hunk 2 above)
Comment (3pointer) [reply pkg/utils/unit.go]: [Done](https://github.com/pingcap/br/pull/118/files#diff-500e3c8287bac6a5d41b94b101065518R52)

File: pkg/restore/client.go [hunk 3]
```diff
@@ -250,7 +252,7 @@ func (rc *Client) RestoreTable(
 	start := time.Now()
 	defer func() {
 		elapsed := time.Since(start)
-		log.Info("Restore Table", zap.Stringer("table", table.Schema.Name), zap.Duration("take", elapsed))
+		summary.Collector.CollectDuration(fmt.Sprintf("restore table: %s", table.Schema.Name), elapsed)
```
Comment (kennytm) [inline pkg/restore/client.go]:
```suggestion
		summary.Collector.CollectDuration("restore table: "+table.Schema.Name, elapsed)
```

File: pkg/restore/client.go [hunk 4]
```diff
@@ -311,7 +313,7 @@ func (rc *Client) RestoreDatabase(
 	start := time.Now()
 	defer func() {
 		elapsed := time.Since(start)
-		log.Info("Restore Database", zap.Stringer("db", db.Schema.Name), zap.Duration("take", elapsed))
+		summary.Collector.CollectDuration(fmt.Sprintf("restore database: %s", db.Schema.Name), elapsed)
```
Comment (kennytm) [inline pkg/restore/client.go]:
```suggestion
		summary.Collector.CollectDuration("restore database: "+db.Schema.Name, elapsed)
```

File: pkg/summary/collector.go [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+func (tc *logCollector) CollectInt(name string, t int) {
+	tc.mu.Lock()
+	defer tc.mu.Unlock()
+	tc.fields = append(tc.fields, zap.Int(name, t))
+}
+
+func (tc *logCollector) Summary(name string) {
+	log.Info(fmt.Sprintf("%s summary", name), tc.fields...)
```
Comment (kennytm) [inline pkg/summary/collector.go]:
```suggestion
	tc.mu.Lock()
	defer tc.mu.Unlock()
	log.Info(name+" summary", tc.fields...)
```

We don't need to clear `tc.fields` after showing the summary?

**Comment by 3pointer** [comment]
Restore:
```
[collector.go:48] ["Restore database summary"] ["restore files"=4] ["restore ranges"=4] ["split region"=870.205229ms] ["restore table: usertable2"=107.958138ms] ["restore table: usertable1"=134.901702ms] ["restore database: br_db"=134.990657ms] ["restore checksum"=6.538697ms]
```

Backup:
```
[collector.go:48] ["Database backup summary"] ["backup total regions"=4] ["backup range finished: start:74800000000000002d5f720000000000000000 end:74800000000000002d5f72ffffffffffffffff00"=88.867504ms] ["backup range finished: start:74800000000000002d5f69800000000000000100 end:74800000000000002d5f698000000000000001fb"=26.644379ms] ["backup range finished: start:74800000000000002f5f720000000000000000 end:74800000000000002f5f72ffffffffffffffff00"=37.044823ms] ["backup range finished: start:74800000000000002f5f69800000000000000100 end:74800000000000002f5f698000000000000001fb"=51.748833ms] ["backup ranges"=206.675391ms] ["backup checksum"=708.394µs]
```

**Comment by 3pointer** [comment]
> This format is a bit bad, you need to refine format of summary log @3pointer
> 
> * please split the information into lines, user will complain to get a very long log if there are many tables
> * please display information according to the structure of the schema and table, it is easier to understand
> * please place the summary information  (like result: sucesss or error message; total time-cost) at the end of the log

split into multiple lines will mix the summary log with other goroutine log, I think put summary log in a line is better. instead of display all schema and tables(or ranges in backup) I put summary log

bakcup success:
```
[2019/12/18 20:15:06.981 +08:00] [INFO] [collector.go:127] ["Full backup summary"] ["backup total regions"=6] ["backup checksum"=1.954634ms] ["total backup ranges: 6, success ranges: 6, failed ranges: 0 total take"=11.82697286s]
```

restore success:
```
[2019/12/18 20:13:17.418 +08:00] [INFO] [collector.go:127] ["Restore database summary"] ["restore files"=4] ["restore ranges"=4] ["split region"=896.129965ms] ["restore checksum"=5.317887ms] ["total restore tables: 2, success tables: 2, failed tables: 0 total take"=217.980366ms]
```

restore failed:
["Restore database summary"] ["restore files"=4] ["restore ranges"=4] ["split region"=884.776009ms] ["total restore tables: 2, success tables: 0, failed tables: 2 failed units"="[usertable2,usertable1]"]

**Comment by GregoryIan** [comment]
please add 
* total kv size(MiB)
* overall throughput(MiB/s) and avg throughput per TiKV instance(MiB/s)
* rename failed case to `failed tables = [schema1.table1, schema2.table2,...]`

**Comment by 3pointer** [comment]
> please add
> 
> * total kv size(MiB)
> * overall throughput(MiB/s) and avg throughput per TiKV instance(MiB/s)
> * rename failed case to `failed tables = [schema1.table1, schema2.table2,...]`

add total kv size from sst_file and avg speed during backup and restore, 
rename failed case 
`backup failed unit -> failed ranges`
`restore failed unit -> failed tables`
now the log looks like:
Restore success:
```
[2019/12/19 11:28:37.102 +08:00] [INFO] [collector.go:167] ["Restore database summary: total restore tables: 2, success tables: 2, failed tables: 0, total take(s): 0.27, total kv: 6, total size(Byte): 199, avg speed(Byte/s): 733.79"] ["restore files"=4] ["restore ranges"=4] ["split region"=899.813178ms] ["restore checksum"=5.766886ms]
```

Backup success:
```
[2019/12/19 11:30:14.544 +08:00] [INFO] [collector.go:167] ["Full backup summary: total backup ranges: 6, success ranges: 6, failed ranges: 0, total take(s): 11.75, total kv: 20000, total size(MB): 10.98, avg speed(MB/s): 0.93"] ["backup total regions"=6] ["backup checksum"=842.938µs]
```

Restore failed:
```
[2019/12/19 11:38:44.138 +08:00] [INFO] [collector.go:142] ["Restore database summary"] ["restore files"=4] ["restore ranges"=4] ["split region"=882.703918ms] ["total restore tables: 2, total success: 0, total failed: 2, failed tables"="[br_db.usertable1,br_db.usertable2]"]
```

**Comment by GregoryIan** [comment]
LGTM

**Pull request merged**